src/RouteResolver.tsx

```diff
@@ -51,22 +39,48 @@ export const getRouteFromParams = (pathname: string) => {
       split.push('')
     }
   }
-  let route = routesConfig.find((el) => split[0] === el.to)!
+  let route = ([...sharedRouteConfig, ...routesConfig] as GenericRoute[]).find(
+    (el) => split[0] === el.to,
+  )!
+
+  if (route == null) {
+    console.warn(`Error: url at ${pathname} has no route mapping`)
+    return [homeRoute, '/']
+  }
+
+  let routePathName = '/' + route.to
+
   // search the route configs for the pathname
   for (let i = 1; i < split.length; i += 1) {
-    if (!route) {
-      return fail(`Error: url at ${pathname} has no route mapping`)
-    }
     if (route.isNested) {
-      route = route.routes.find((el) => el.to!.includes(split[i]))!
-    } else {
-      fail(`Route at ${pathname} has no SynapseConfigArray mapping`)
+      const nextRoute = route.routes.find((el) => el.to!.includes(split[i]))
+      if (nextRoute) {
+        route = nextRoute
+        routePathName += '/' + route.to
+      } else {
+        // If we can't find a matching nested route, return the last route that was found
+        break
+      }
     }
   }
```

Comment:
Gross loop that involves a `break` statement. Might be easier to understand if you look at the test cases

Reply:
Should probably output a warning to the console about the sub-route that can't be found.